src/Illuminate/Contracts/Auth/Guard.php

```diff
@@ -21,7 +21,7 @@ public function guest();
     /**
      * Get the currently authenticated user.
      *
-     * @return \Illuminate\Contracts\Auth\Authenticatable|null
+     * @return \Illuminate\Contracts\Auth\Authenticatable|\App\User|null
```

Comment:
If you use a `App\User` class - but use a different class for Authenticating - that scenario would give the wrong autocomplete.

Anyway - regardless - its not about side effects - its about hard coding an App variable into a framework that should not have knowledge of it. That is why the contracts exists.